spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/datastore/it/GcpDatastoreEmulatorIntegrationTests.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.datastore.it;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.auth.Credentials;
+import org.junit.Test;
+
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.DatastoreRepositoriesAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.DatastoreTransactionManagerAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.GcpDatastoreAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.datastore.GcpDatastoreEmulatorAutoConfiguration;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTemplate;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.annotation.Id;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for Datastore Emulator integration with the datastore itself.
+ *
+ * @author lucasoares
+ */
+public class GcpDatastoreEmulatorIntegrationTests {
+	@Test
+	public void testSaveDatastoreEntity() {
+		new ApplicationContextRunner()
+				.withConfiguration(AutoConfigurations.of(GcpDatastoreAutoConfiguration.class,
+						GcpContextAutoConfiguration.class,
+						DatastoreTransactionManagerAutoConfiguration.class,
+						DatastoreRepositoriesAutoConfiguration.class,
+						GcpDatastoreEmulatorAutoConfiguration.class))
+				.withUserConfiguration(TestConfiguration.class)
+				.withPropertyValues("spring.cloud.gcp.datastore.project-id=test-project",
+						"spring.cloud.gcp.datastore.namespace=test-namespace",
+						"spring.cloud.gcp.datastore.emulator-host=localhost:8181",
+						"spring.cloud.gcp.datastore.auto-emulator.port=8181",
+						"spring.cloud.gcp.datastore.auto-emulator.enabled=true",
+						"spring.cloud.gcp.datastore.auto-emulator.consistency=0.9")
+				.run((context) -> {
+					DatastoreTemplate datastore = context.getBean(DatastoreTemplate.class);
+
+					EmulatorEntityTest entity = new EmulatorEntityTest();
+					entity.setProperty("property-test");
+
+					datastore.save(entity);
+
+					assertThat(entity.getId()).isNotNull();
+
+					assertThat(datastore.findById(entity.getId(), EmulatorEntityTest.class).getProperty())
```

Comment:
I will look into that.

Reply:
I don't think I can do that, not easly anyway.. The only way to do that is manually making a GET request to the localhost:<PORT>/<some_path_that_I_need_to_search> to ping the emulator.

If you think I should do that, I can look and search for a specific request to test if the emulator is running. 

I will start checking for the running parameter at the auto configuration class, its a good start to check the `stop()` method execution since its now a smart lifecycle.